From: Maximilian Weigand <mweigand@mweigand.net>
Date: Tue, 25 Jul 2023 21:36:15 +0200
Subject: [rockchip_ebc] add a new module parameter "temp_override" that lets
 you fix the assumed panel temperature. A value of -1 will use the actual
 panel temperature

---
 drivers/gpu/drm/rockchip/rockchip_ebc.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/rockchip/rockchip_ebc.c b/drivers/gpu/drm/rockchip/rockchip_ebc.c
index 7923440..ebe7902 100644
--- a/drivers/gpu/drm/rockchip/rockchip_ebc.c
+++ b/drivers/gpu/drm/rockchip/rockchip_ebc.c
@@ -252,6 +252,11 @@ static bool prepare_prev_before_a2 = false;
 module_param(prepare_prev_before_a2, bool, 0644);
 MODULE_PARM_DESC(prepare_prev_before_a2, "Convert prev buffer to bw when switchting to the A2 waveform");
 
+static int temp_override = 0;
+module_param(temp_override, int, S_IRUGO|S_IWUSR);
+MODULE_PARM_DESC(temp_override, "Values > 0 override the temperature");
+
+
 DEFINE_DRM_GEM_FOPS(rockchip_ebc_fops);
 
 static int ioctl_trigger_global_refresh(struct drm_device *dev, void *data,
@@ -1072,6 +1077,11 @@ static void rockchip_ebc_refresh(struct rockchip_ebc *ebc,
 		/* Convert from millicelsius to celsius. */
 		temperature /= 1000;
 
+		if (temp_override > 0){
+			printk(KERN_INFO "rockchip-ebc: override temperature from %i to %i\n", temp_override, temperature);
+            temperature = temp_override;
+        }
+
 		ret = drm_epd_lut_set_temperature(&ebc->lut, temperature);
 		if (ret < 0)
 			drm_err(drm, "Failed to set LUT temperature: %d\n", ret);
